odoo/addons/base/tests/test_res_partner.py

```diff
@@ -138,3 +138,18 @@ def test_lang_computation_form_view(self):
         self.assertEqual(partner.lang, 'fr_FR')
         self.assertEqual(partner.child_ids.filtered(lambda p: p.name == "First Child").lang, 'de_DE')
         self.assertEqual(partner.child_ids.filtered(lambda p: p.name == "Second Child").lang, 'fr_FR')
+
+    def test_display_address_missing_key(self):
+        """ Check _display_address when some keys are missing. As a defaultdict is used, missing keys should be
+        filled with empty strings. """
+        country = self.env["res.country"].create({
+            "name": "TestCountry",
+            "address_format": "%(city)s %(zip)s %(nothing)s",
+        })
+        partner = self.env["res.partner"].create({
+            "name": "TestPartner",
+            "country_id": country.id,
```

Comment:
You can't use `create` because it checks the placeholders.
```suggestion
        self.env.cr.execute("UPDATE res_country SET address_format ='%(city)s %(zip)s %(nothing)s' WHERE id=1")
        partner = self.env["res.partner"].create({
            "name": "TestPartner",
            "country_id": 1,
```

Reply:
It won't cause trouble to other test if we modify the data ?
Isn't it better to create a `TestCountry` and update the `address_format` ?

Reply:
I don't think this will cause issues (@KangOl ?), but to be sure you could create your dummy country and set the address_format in a query like done here. Just don't pass the format in the create method (or through the ORM ;))